### PR TITLE
Make htmlparser parse unquoted attrib values

### LINKED ISSUE #6154: htmlparser breaks on unquoted attributes
HTML5 allows attributes to be unquoted, but htmlparser can't handle it.

F.ex 
```html
<!DOCTYPE html>
<html>
    <head>
      <title> foobar </title>
    </head>
    <body>
      <p class=foo id=bar></p>
    </body>
</html>
```

is parsed as
```html
<document>
<html>
    <head>
      <title> foobar </title>
    </head>
    <body>
      <p class="foo">id=bar&gt;</p>
    </body>
</html>
</document>
```

### PR DESCRIPTION
**Pull request by recloser**
Fix #6154.

Implemented in accordance to https://html.spec.whatwg.org/multipage/syntax.html#unquoted

### CODE REVIEW

File: lib/pure/parsexml.nim [hunk 1]
```diff
@@ -669,6 +672,20 @@ proc parseAttribute(my: var XmlParser) =
             pendingSpace = false
           add(my.b, buf[pos])
           inc(pos)
+  elif allowUnquotedAttribs in my.options:
+    const disallowedChars = "\"'`=<>\0\t\L\F\f "
```
Comment (Araq) [inline lib/pure/parsexml.nim]: Use a set here.

Comment (recloser) [reply lib/pure/parsexml.nim]: Ok.

File: lib/pure/parsexml.nim [hunk 2]
```diff
@@ -669,6 +672,21 @@ proc parseAttribute(my: var XmlParser) =
             pendingSpace = false
           add(my.b, buf[pos])
           inc(pos)
+  elif allowUnquotedAttribs in my.options:
+    const disallowedChars = {'"', '\'', '`', '=', '<', '>', ' ',
+                             '\0', '\t', '\L', '\F', '\f'}
+    let startPos = pos
+    while (let c = buf[pos]; c notin disallowedChars):
```
Comment (dom96) [inline lib/pure/parsexml.nim]: You should probably protect against an IndexError here.

Comment (recloser) [reply lib/pure/parsexml.nim]: Nah, it's consistent with the rest of the code which just checks for `'\0'`. Nothing should `inc` past EOF.

**Comment by Araq** [comment]
Unrelated test failures, merging.

**Pull request merged**